src/Nethermind/Nethermind.Blockchain/IBlockTree.cs

```diff
@@ -130,6 +130,8 @@ public interface IBlockTree : IBlockFinder
         BlockHeader[] FindHeaders(Keccak hash, int numberOfBlocks, int skip, bool reverse);
         
         void DeleteInvalidBlock(Block invalidBlock);
+
+        void Flush();
```

Comment:
BlockTree alredy has a polluted API, this one seems very internal

Reply:
Moved flush on demand when read is done